### ISSUE TITLE
Input Capture Requested dialog needs to be clicked afgter every restart

### ISSUE THREAD

**Issue by mistyron** [input-leap/input-leap#1979]
### What happened?

I want to start Input Leap (server) on Login and I want that clients can connect to it right away but that sdoesn't work. Instead I have to open the Input Leap dialog, click reload upon which the diaklog "Input Capture Requested" shows up and I need to swelect Allow every time. Is there a way that I can start Input Leap with a flag so taht it automatically allows connection? It would be the same thing on the client side as well.

### Version

v2.4.0

### If applicable, where did you install Input Leap from?

Fedora Repositories

### What OSes are you seeing the problem on? (Check all that apply)

Linux

### If applicable, are you using Wayland or X11?

Wayland

### What OS versions are you using?

Fedora Linux 40

**Comment by nbolton** [comment]
Yeah the permission dialog is pretty annoying. I plan to do something about this. Not sure what is possible yet.

@whot Any ideas?

**Comment by sithlord48** [comment]
I would suspect this is going to be a issue that has to be fixed within the xdg-portal backend .

**Comment by whot** [comment]
https://github.com/flatpak/xdg-desktop-portal/pull/1186 - needs to be implemented in the compositors though which is where I got stuck due to ETIME

**Comment by shymega** [comment]
I discussed this with @nbolton over a call.

My vision would be the following flow:

- Software (let's say Input Leap) starts up.
- Software verifies the SSL fingerprint of screen(/s) and stores the fingerprint.
- The portal bus interface is requested. The user is presented with an 'Always trust this session' checkbox, which is then stored with the portal.
- If the user checks this box, the portal will continue the flow as it does currently but inform the portal caller of the 'Trust' checkbox for future use.

Now, let's say, the user starts up Input Leap again. We go through the flow again, but in this case, the user has checked the 'Always trust' checkbox previously. 

We then verify the SSL fingerprint has not changed. If it has, we should ask the user for permission again, and continue the flow, as described above.

In an alternate scenario, the SSL fingerprint hasn't changed, _and_ the user has checked the 'Always trust' checkbox before. We don't want to ask the user again for permission because they have indicated they trust the connection. This would be bad UI/UX design.

In this case, the portal knows the user is OK with the connection, and the SSL fingerprint is valid. Input Leap then continues the connection sequence, and we're running.

The only problem with this is when SSL is disabled. We should think about this.

I'd welcome any thoughts on the flow that I described to Nick over our call.

**Comment by whot** [comment]
A note here: the portal has a [permission store](https://github.com/flatpak/xdg-desktop-portal/wiki/The-Permission-Store) so this is the first point to look at. Once that's added for the RD and IC portals the popups should go away altogether. Adding it shouldn't be that hard (technically).

Mind you, the SSL check does not need to/should not be in the portal interaction - that one is for *local* permissions only with the idea that if the user says "input-leap is fine" then it can always connect. The SSL verification for remote servers/clients should then happen in input-leap and rejected there.

**Comment by sithlord48** [comment]
This is an upstream bug and we have this tracked in our Wayland Discussion https://github.com/input-leap/input-leap/discussions/1976

**Comment by nbolton** [comment]
Relavent PR: 
- https://github.com/deskflow/deskflow/pull/8865

<img width="782" height="622" alt="Screenshot From 2025-08-19 09-53-04" src="https://github.com/user-attachments/assets/05aed9de-32ff-45aa-9016-707c3f35046d" />
